Src/Base/AMReX_FBI.H

```diff
@@ -54,19 +54,26 @@ ParallelFor (Vector<Array4BoxTag<T> > const& tags, int ncomp, F && f)
     }
     nwarps.push_back(ntotwarps);
 
-    std::size_t nbytes = ntags*sizeof(TagType);
-    auto d_tags = static_cast<TagType*>(The_Device_Arena()->alloc(nbytes));
-    Gpu::htod_memcpy(d_tags, tags.data(), nbytes);
+    std::size_t sizeof_tags = ntags*sizeof(TagType);
+    std::size_t offset_nwarps = Arena::align(sizeof_tags);
+    std::size_t sizeof_nwarps = (ntags+1)*sizeof(int);
+    std::size_t total_buf_size = offset_nwarps + sizeof_nwarps;
 
-    nbytes = (ntags+1)*sizeof(int);
-    auto d_nwarps = static_cast<int*>(The_Device_Arena()->alloc(nbytes));
-    Gpu::htod_memcpy(d_nwarps, nwarps.data(), nbytes);
+    char* h_buffer = (char*)The_Pinned_Arena()->alloc(total_buf_size);
+    char* d_buffer = (char*)The_Arena()->alloc(total_buf_size);
 
-    constexpr int nthreads = 128;
+    std::memcpy(h_buffer, tags.data(), sizeof_tags);
+    std::memcpy(h_buffer+offset_nwarps, nwarps.data(), sizeof_nwarps);
+    Gpu::htod_memcpy_async(d_buffer, h_buffer, total_buf_size);
+
+    auto d_tags = reinterpret_cast<TagType*>(d_buffer);
+    auto d_nwarps = reinterpret_cast<int*>(d_buffer+offset_nwarps);
+
+    constexpr int nthreads = 256;
```

Comment:
Was the increase in nthreads tested at all? Or just changing to be consistent with other launches?

Should this be based on MAX_NUM_THREADS or another more universal variable, so it can be easily adjusted for each architecture type without being forgotten?

Reply:
Yes, it was tested.  But I didn't really see much difference.